packages/pangea-node-sdk/src/services/base.ts

```diff
@@ -20,15 +20,39 @@ class BaseService {
   */
   constructor(
     serviceName: string,
-    token: string,
+    token: PangeaToken,
     config: PangeaConfig,
     configID?: string
   ) {
     if (!serviceName) throw new Error("A serviceName is required");
     if (!token) throw new Error("A token is required");
 
     this.serviceName = serviceName;
-    this.token = token;
+    if (typeof token === "string") {
```

Comment:
Wouldn't be better to move this whole logic into a function?

Reply:
It's only used here, we can break it out later if more use comes. I'm not a fan of early abstraction, generally abstract only a rule of 3.